search.php

```diff
@@ -45,11 +121,21 @@
             echo "<a class=img-result href='/image_proxy.php?url=", $url, "'>";
             echo "<img src='/image_proxy.php?url=", $url, "'></a>";
         }
-        return $images;
     echo "</div>";
+    $result = new SearchResult();
+    $result->images = $images;
+    if (property_exists($data->{"resource_response"}, "bookmark"))
+        $result->bookmark = $data->{"resource_response"}->{"bookmark"};
+    return $result;
 };
 
-$images = $search("$query");
+$result = $search($query, $bookmark);
+
+if ($result->bookmark != null)
+{
+    $bookmark_encoded = urlencode($result->bookmark);
+    echo "<h2 style=\"text-align: center;\"><a href=\"/search.php?q=$query&bookmark=$bookmark_encoded&csrftoken=$csrftoken\">Next page</a></h2><br><br><br>";
```

Comment:
It seems that this is XSS vulnerable 🤔

Reply:
I tried searching `"><script>alert(1)</script>` and it succeeded.

Reply:
Fixed, thank you!